solutions/gold/usaco-694.mdx

```diff
@@ -165,5 +165,57 @@ public class HoofPaperScissors {
 }
 ```
 </JavaSection>
+<PySection>
 
+<Warning>
+
+Due to Python's constant factor, the below solution TLEs on a couple test cases.
+
+</Warning>
+
+```py
+MX = 10**5 + 5
+
+dp = [[[0] * 3 for _ in range(25)] for _ in range(MX)]
+A = [0] * MX
+
+with open("hps.in") as read:
+	n, k = map(int, read.readline().strip().split())
+
+	for i in range(n):
+		a = read.readline().strip()
+		if a == "H":
+			A[i] = 0
+		elif a == "P":
+			A[i] = 1
+		else:
+			A[i] = 2
+
+# either she switches to h or p or s or stays
+for i in range(n):
+	for j in range(k + 1):
+		for l in range(3):
+			if l == A[i]:
+				dp[i][j][l] += 1
+
+			# switch to not item
```

Comment:
could you name the items? (hoof, paper, scissors)

Reply:
tell me if the current one is right.